applet/src/radiostate.c

```diff
@@ -23,13 +24,26 @@ int rst_hdr_dst ;
 
 void rst_voice_lc_header(int src, int dst, int groupcall)
 {
+    #define BSIZE 100
+    char src_buf[BSIZE];
+    char dst_buf[BSIZE];
+
     if( !rst_voice_active || rst_src != src || rst_dst != dst) {
         rst_src = src ;
         rst_dst = dst ;
+
         PRINT("\n* Call from %d to %s%d started.\n", src, groupcall ? "group ":"", dst);
         LOGR("cs %d->%s%d\n", src, groupcall ? "group ":"", dst );
+
+        if( find_dmr_user(src_buf, src, (void *) 0x100000, BSIZE) ) {
```

Comment:
can you move the lookup to the display thread? you can store the dmrids in the interrupt handler, and lookup them up in the display thread, and print them to the log.